data-management/data-interface.js

```diff
@@ -324,22 +324,15 @@ const revokeAccess = async (parameters, context) => {
     }
 }
 
-const disableAdmin = (user, params) => {
+const disableAdmin = async (userID, params, context) => {
+    const aUser = await getUser({userID: userID}, context);
     const userParams = {};
-    const aUser = JSON.parse(JSON.stringify(user));
-    if (isCaseInsensitiveEqual(aUser.role, ADMIN) && params.role !== ADMIN) {
-        const prevRole = !aUser.prevRole ? MEMBER : aUser.prevRole;
-        if (isCaseInsensitiveEqual(prevRole, MEMBER)) {
-            userParams.userStatus = ACTIVE;
-        }
-        // userStatus is inactive when userStatus is not specified and the user has no approved acl
-        const isNoneStatusUser = aUser.userStatus === NONE || !aUser.userStatus;
+    const isAdminUser = isCaseInsensitiveEqual(aUser.role, ADMIN);
+    const isAdminRevoke = !isCaseInsensitiveEqual(params.role, ADMIN);

```

Comment:
When params.role = null (not passed in as a parameter) this will trigger isAdminRevoke to be true. We should consider updating this to check if isCaseInsensitiveEqual(params.role, MEMBER) || isCaseInsensitiveEqual(params.role, NON_MEMBER);